src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -146,19 +147,25 @@ public void start(Promise<Void> startPromise) {
+                if (this.bridgeConfig.getHttpConfig().isProducerEnabled()) {
+                    routerBuilder.operation(this.SEND.getOperationId().toString()).handler(this.SEND);
+                }
                 routerBuilder.operation(this.SEND_TO_PARTITION.getOperationId().toString()).handler(this.SEND_TO_PARTITION);
-                routerBuilder.operation(this.CREATE_CONSUMER.getOperationId().toString()).handler(this.CREATE_CONSUMER);
-                routerBuilder.operation(this.DELETE_CONSUMER.getOperationId().toString()).handler(this.DELETE_CONSUMER);
-                routerBuilder.operation(this.SUBSCRIBE.getOperationId().toString()).handler(this.SUBSCRIBE);
-                routerBuilder.operation(this.UNSUBSCRIBE.getOperationId().toString()).handler(this.UNSUBSCRIBE);
-                routerBuilder.operation(this.LIST_SUBSCRIPTIONS.getOperationId().toString()).handler(this.LIST_SUBSCRIPTIONS);
-                routerBuilder.operation(this.ASSIGN.getOperationId().toString()).handler(this.ASSIGN);
-                routerBuilder.operation(this.POLL.getOperationId().toString()).handler(this.POLL);
-                routerBuilder.operation(this.COMMIT.getOperationId().toString()).handler(this.COMMIT);
-                routerBuilder.operation(this.SEEK.getOperationId().toString()).handler(this.SEEK);
-                routerBuilder.operation(this.SEEK_TO_BEGINNING.getOperationId().toString()).handler(this.SEEK_TO_BEGINNING);
-                routerBuilder.operation(this.SEEK_TO_END.getOperationId().toString()).handler(this.SEEK_TO_END);
+                if (this.bridgeConfig.getHttpConfig().isConsumerEnabled()) {
+                    routerBuilder.operation(this.CREATE_CONSUMER.getOperationId().toString()).handler(this.CREATE_CONSUMER);
+                    routerBuilder.operation(this.DELETE_CONSUMER.getOperationId().toString()).handler(this.DELETE_CONSUMER);
+                    routerBuilder.operation(this.SUBSCRIBE.getOperationId().toString()).handler(this.SUBSCRIBE);
+                    routerBuilder.operation(this.UNSUBSCRIBE.getOperationId().toString()).handler(this.UNSUBSCRIBE);
+                    routerBuilder.operation(this.LIST_SUBSCRIPTIONS.getOperationId().toString()).handler(this.LIST_SUBSCRIPTIONS);
+                    routerBuilder.operation(this.ASSIGN.getOperationId().toString()).handler(this.ASSIGN);
+                    routerBuilder.operation(this.POLL.getOperationId().toString()).handler(this.POLL);
+                    routerBuilder.operation(this.COMMIT.getOperationId().toString()).handler(this.COMMIT);
+                    routerBuilder.operation(this.SEEK.getOperationId().toString()).handler(this.SEEK);
+                    routerBuilder.operation(this.SEEK_TO_BEGINNING.getOperationId().toString()).handler(this.SEEK_TO_BEGINNING);
+                    routerBuilder.operation(this.SEEK_TO_END.getOperationId().toString()).handler(this.SEEK_TO_END);
+                }
```

Comment:
Much cleaner way would be to return some nice and self-explanatory error when it is not enabled. And not just not-build the route.

Reply:
Thanks for the review, I will do the necessary changes but can you please suggest the location where I can do the above, I am very new to java and this is my first open source commit.

Reply:
updated the position of consumer/producer disable/enable as suggested by @ppatierno